### DISCUSSION TITLE
Non-SaaS/Local search provider?

### DISCUSSION THREAD

**Discussion by wayneyaoo** [Lednerb/bilberry-hugo-theme#616] [Ideas]
Hi huge thanks for making the great theme with the neat idea of "post types"! But is there any possibility that we can go from service-based search to more privacy-focused search mechanism like https://www.fusejs.io/ simple local search? Using Algolia would require every search to do an external post which is a bit... weird for a "static site" :).

I'm no expert in JS and could underestimate the effort to adopt another library though.

Thanks again for the nice theme!

**Comment by wayneyaoo** [comment]
Oh sorry I did a few more search and this ended up being a duplicate of #552 . Is implementing this feature on our own like recommended in the discussion still true today?

**Reply by igor-baiborodine** [reply]
@wayneyaoo, yes, implementing integration with the `fuse.js` library should be similar to the one I suggested for `lunr.js`.
